### ISSUE TITLE
return Result instead of panic

### ISSUE THREAD

**Issue by xemwebe** [fbriden/yahoo-finance-rs#13]
Hi, thank's for writing this little library, saves me from doing it myself... ;-) 

However, I observed that for some valid ticker, the library panics, crashing my application which is based on it. Here is an example:
```rust
use yahoo_finance::history;
use std::time::{Duration, UNIX_EPOCH};
use chrono::{DateTime,Utc};

fn main() {
   let data = history::retrieve("ISPFF").unwrap();
   for bar in &data {
      println!("On {} ISPFF closes at ${:.2}", 
          DateTime::<Utc>::from(UNIX_EPOCH + Duration::from_secs(bar.timestamp)), bar.close)
   }
}
```
Apparently, the cause is that yahoo delivers a null value in field `firstTradeDate`, which causes the `unwrap` in line 31 of `chart.rs` to panic:
```rust
let response: Response = serde_json::from_str(x).unwrap();
```
Could you  consider to return a `Result` instead of letting the library panic? In this case, the application calling the library could decide how to deal with it. Of course, you could just avoid using these ticker (which is recommended anyway since yahoo delivers only sparse data for that one), but I would prefer if I could check that without testing for panic.

**Comment by fbriden** [comment]
Good feedback!  Let me add to the list for v0.3

**Comment by kov** [comment]
I saw that the more recent code seems to have somewhat addressed this:

```rust
   let chart = serde_json::from_str::<Response>(&data).context(error::BadData)?.chart;
```

However, I hit an issue where my code queries for data on days the market was closed on but I only get back a generic error. This was the URL hit, notice the quote array has an empty object.

https://query1.finance.yahoo.com/v8/finance/chart/ANIM3.SA?period1=1598659200&period2=1598842799&interval=1d

It would be important for my code to be able to distinguish between 'no data', which is the case there, and actual failure. The way the error code works at the moment I can't see that what I got was an error of type BadData and assume it was just no data (to distinguish it from network errors for instance).

I am new to Rust and can't think of a great solution, but I did notice you used to have Error as an enum. That could be a good way of letting the client get some details other than for the message. Any reason you moved away from that? Also, should I open a separate issue?
